src/jit/function-builder.cc

```diff
@@ -195,14 +198,43 @@ bool FunctionBuilder::buildIL() {
   return true;
 }
 
-void FunctionBuilder::SetUpLocals(TR::IlBuilder* b, const uint8_t** pc, VirtualStack* stack) {
+bool FunctionBuilder::SetUpLocals(TR::IlBuilder* b, const uint8_t** pc, VirtualStack* stack) {
   // Add placeholders onto the virtual stack. These values should never be actually read, but will
   // eventually be dropped using drop or drop_keep in the function epilogue.
+  static const auto supported_types = std::unordered_set<Type>({Type::I32, Type::I64, Type::F32, Type::F64});
   for (size_t i = 0; i < fn_->param_and_local_types.size(); i++) {
-    stack->Push(nullptr);
+    auto t = fn_->param_and_local_types.at(i);
+    if (supported_types.end() == supported_types.find(t))
+        return false;
+
+    auto locals_offset = fn_->param_and_local_types.size() - fn_->local_count;
+    if (i < locals_offset) {
+      // local is parameter, so we load it from the physical stack
+      auto param_addr = PickPhys(b, locals_offset - i);
+      auto param = b->LoadIndirect("Value", TypeFieldName(t), param_addr);
+      stack->Push(param);
+    } else {
+      switch (t) {
+          case Type::I32:
+            stack->Push(b->ConstInt32(0));
+            break;
+          case Type::I64:
+            stack->Push(b->ConstInt64(0));
+            break;
+          case Type::F32:
+            stack->Push(b->ConstFloat(0.0));
+            break;
+          case Type::F64:
+            stack->Push(b->ConstDouble(0.0));
+            break;
+          default:
+            TR_ASSERT_FATAL(false, "Local %d is of unexpected type %d\n", i, t);
```

Comment:
I've chanced this back to an assert because the check at the beginning of the loop should prevent us from ever reaching this branch. If we somehow do, then something is _very_ wrong.